hdat/hdat_cli.py

```diff
@@ -76,49 +78,100 @@ def hdat_cli(arguments, suites, golden_store, archive, git_info):
+            results = resolve_resultspecs(archive, suites, args.resultspec)
             for result in results:
-                show_result(suites, result)
+                exit_status = show_result(suites, result)
+                if exit_status == 1:
+                    exit_code = 1
         for resultspec in args.resultspec:
-            results = resolve_resultspecs(archive, resultspec)
-            for result in results:
-                show_result(suites, result)
+            results = resolve_resultspecs(archive, suites, resultspec)
+            if isinstance(results, str):
+                exit_code = 1
+                print(results)
+            else:
+                for result in results:
+                    if isinstance(results, str):
+                        exit_code = 1
+                        print(result)
+                    else:
+                        exit_status = show_result(suites, result)
+                        if exit_status == 1:
+                            exit_code = 1
     elif args.command == 'runshow':
         cases = resolve_casespecs(suites, args.casespecs)
         cases_status = run_cases(suites, golden_store, archive, git_info, cases)
         if cases_status['error'] > 0:
             raise AbortError(_format_cases_status(cases_status))
         for casespec in args.casespecs:
-            results = resolve_resultspecs(archive, casespec)
+            results = resolve_resultspecs(archive, suites, casespec)
             for result in results:
-                show_result(suites, result)
+                if isinstance(results, str):
+                    exit_code = 1
+                    print(result)
+                else:
+                    exit_status = show_result(suites, result)
+                    if exit_status == 1:
+                        exit_code = 1
     elif args.command == 'diff':
-        golden_results = resolve_resultspecs(archive, args.resultspec[0])
-        results = resolve_resultspecs(archive, args.resultspec[1])
-        for golden_result, result in zip(golden_results, results):
-            diff_results(suites, golden_result, result)
+        golden_results = resolve_resultspecs(archive, suites, args.resultspec[0])
+        results = resolve_resultspecs(archive, suites, args.resultspec[1])
+        if isinstance(golden_results, str) or isinstance(results, str):
+            exit_code = 1
+            if isinstance(golden_results, str):
+                print(golden_results)
+            if isinstance(results, str):
+                print(results)
+        else:
+            for golden_result, result in zip(golden_results, results):
+                exit_code = diff_results(suites, golden_result, result)
     elif args.command == 'verify':
-        results = resolve_resultspecs(archive, args.resultspec)
-        for result in results:
-            golden_store.insert(result)
+        results = resolve_resultspecs(archive, suites, args.resultspec)
```

Comment:
Unlike `hdat show`, we want `hdat verify` to be all or nothing.  Unlike `hdat show`, which involves user interactions and which could be annoying if we fail half-way through, `hdat verify` doesn't require any user interaction.  Also, it is (or should be) a fast operation.  Thus it is best for it to be all or nothing.